app/reaction.go

```diff
@@ -105,11 +111,12 @@ func (a *App) sendReactionEvent(event string, reaction *model.Reaction, post *mo
 	message.Add("reaction", reaction.ToJson())
 	a.Publish(message)
 
-	// The post is always modified since the UpdateAt always changes
-	a.InvalidateCacheForChannelPosts(post.ChannelId)
 	post.HasReactions = hasReactions
 	post.UpdateAt = model.GetMillis()
+
+	clientPost := a.PreparePostForClient(post)
```

Comment:
The `post_update` websocket event is fired to update `post.HasReactions` which is used to tell the client it should load reactions when first rendering a given post. It shouldn't be necessary any more since it'll receive those reactions as part of the metadata (hence the stuff about that field being deprecated), but we still need to keep this for at least a few releases to support older mobile apps on the new server.

To answer your question, the client that makes the reaction may wait for it, but this is primarily needed for clients that have the post but not the reactions loaded. We could consider just regenerating some parts of the post, but I don't know how the client would handle that at this time.

As for any possible race conditions, this may be adding some, but I'd bet those probably exist already.

Reply:
Also, I rewrote the first paragraph of that comment repeatedly since I keep going back and forth in my head about whether or not the `update_post` event is actually needed for older clients in the first place since it solves a very specific scenario that I'm not sure if we can actually hit because of how the client loads posts. I'll think more about that since it would be nice to remove that event sooner rather than later